services/telemetry/telemetry.go

```diff
@@ -149,13 +149,13 @@ func (ts *TelemetryService) getRudderConfig() RudderConfig {
 	}
 }
 
-func (ts *TelemetryService) telemetryEnabled() bool {
+func (ts *TelemetryService) TelemetryEnabled() bool {
 	return *ts.srv.Config().LogSettings.EnableDiagnostics && ts.srv.IsLeader()
```

Comment:
I don't think we can use `TelemetryEnabled` **for the API endpoints** because it checks both an application setting **and** status within a cluster of Mattermost servers. In clustered deployments, a client can get inconsistent answers about whether true-up is/should be performed automatically.

So for example, if a customer's deployment has 2 servers to give them high availability, then depending on how load balancing is handled, the user may:
* hit the follower for the get true up status
* be told that telemetry is not enabled
* see and click the true up submit button
* the request is sent to the leader
* the leader fails to update the request since telemetry is enabled.
* The user doesn't see any change of state occur since the endpoint returns that status and doesn't send or complete it.

I think we'll need to check the EnableDiagnostics setting instead. I don't see another helper for checking EnableDiagnostics. 0/5 on whether we want it.

We may not need to return `telemetry_enabled` to the client. It looks like users have access to that config information already:
* definition: https://github.com/mattermost/mattermost-webapp/blob/master/packages/types/src/config.ts#L62
* endpoint /api/v4/config/client?format=old'
* 
It should be accessible through the getConfig selector.

I think we **do** want to use `TelemetryEnabled()` **for the job**, because that way in clustered deployments, we don't get multiple nodes trying to send the true up information.

Reply:
Make sense. I'll make this change shortly, just running out to get something to eat for lunch.